### DISCUSSION TITLE
Request to port and maintain simplify-commerce

### DISCUSSION THREAD

**Discussion by bohan0** [garronej/denoify#65] [General]
Hi

I need to use https://www.npmjs.com/package/simplify-commerce in Deno (specifically, in Supabase Edge Functions which run on top of Deno Deploy - https://supabase.com/docs/guides/functions ). I would like to pay someone to create and maintain the Denoify port and publish it on deno.land/x and Github for all to benefit.

This library from Mastercard is updated very rarely (only a single new version in last 4 years!) so would not be a lot of work to maintain for someone with experience, it also only has one external dependency (jws).

I am willing to pay any reasonable amount to get this done quickly. I need to use this payment gateway in our app and do not have the time/skill/experience learn and use Denoify properly at the moment.

**Comment by garronej** [comment]
I've looked into it. `simplify-commerce` is denoifyiable but first why don't you try experimental npm support by Deno: https://deno.com/blog/v1.25#experimental-npm-support

**Comment by bohan0** [comment]
The problem is, I cannot control the version of Deno being used by Supabase Edge Functions ( https://github.com/supabase/supabase/discussions/8006 ). Also, I think Supabase will not be supporting unstable Deno features. So it could be many months before I can try the import "npm:simplify-commerce" method.

**Comment by garronej** [comment]
WIP: https://github.com/garronej/simplify-commerce

**Reply by garronej** [reply]
I hoped I'd be able to use https://deno.land/x/djwt@v2.7 in place of https://www.npmjs.com/package/jws with a `jws.deno.ts` but djwt is async while jws is sync.  
I'll have to denoify the jws...

**Reply by garronej** [reply]
I'll try to do it when I can but if I don't want to set you back. If you find someone else to do it go ahead.

**Reply by bohan0** [reply]
Please try and do it in your spare time, I am happy to wait. Thanks again for your time and effort :)
I have a feeling that jws may be a part of the ~10% of packages that will not import using "npm:X" via deno 1.25+ anyway so it should be helpful for many.

**Comment by bohan0** [comment]
As of esm.sh v99 and node-std v0.167, simplify-commerce is now working via
CDN :)

**Reply by garronej** [reply]
Great sory I didn't find the time to deliver and thanks for the update!
